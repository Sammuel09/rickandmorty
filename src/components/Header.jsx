import { useState, useEffect } from "react";
import rickAndMortyLogo from "../assets/images/rickandmorty.svg";
import clsx from "clsx";

const Header = () => {
  const [extraClass, setExtraClass] = useState("solid");

  const handleScroll = () => {
    const show = window.scrollY > 310;
    if (show) {
      setExtraClass("transparent");
    } else {
      setExtraClass("solid");
    }
  };
  useEffect(() => {
    document.addEventListener("scroll", handleScroll);

    return () => {
      document.removeEventListener("scroll", handleScroll);
    };
  }, []);
  return (
    <div
      className={clsx({
        ["header"]: true,
        [extraClass]: true,
      })}
    >
      <img src={rickAndMortyLogo} alt="Rick and Morty Logo" />
    </div>
  );
};

export default Header;
