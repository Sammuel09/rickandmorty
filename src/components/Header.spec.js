import React from "react";
import { render, cleanup } from "@testing-library/react";
import Header from "./Header";

afterEach(cleanup);
it("should take a snapshot", () => {
  const { asFragment } = render(<Header />);
  expect(asFragment(<Header />)).toMatchSnapshot();
});
