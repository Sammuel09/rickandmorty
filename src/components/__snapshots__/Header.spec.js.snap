// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <img
      alt="Rick and Morty Logo"
      src="rickandmorty.svg"
    />
  </div>
</DocumentFragment>
`;
